src/ngraph/ngraph_emitter.cc

```diff
@@ -1365,6 +1365,25 @@ void Emitter::CreateLayerOps() {
     return ngraph::builder::make_with_numpy_broadcast<ngraph::op::Add>(
         convolution, bias_reshape);
   };
+  ngraph_op_funcs_["Deconvolution"] =[this](const NodePtr& node) -> NgraphNodePtr {
+    NgraphNodePtr data = op_map_[node->inputs_[0]];
+    NgraphNodePtr filter = op_map_[node->inputs_[1]];
+
+    // N, channel_in, d1,...,dn
+    const auto data_shape = data->get_shape();
+    // channel_out, channel_in/groups, f1,...,fn
+    const auto filter_shape = filter->get_shape();
+    const auto out_shape = TShape_to_NShape(node->shape_);
+
+    auto n = data_shape.size() - 2;
+    auto pad = get_default<ptrdiff_t>(node, "pad", ngraph::CoordinateDiff(n, -1));
+    auto stride = get_default<size_t>(node, "stride", ngraph::Strides(n,1));
+    auto dilate = get_default<size_t>(node, "dilate", ngraph::Strides(n,1));
```

Comment:
I notice that MXnet's Deconvolution operator has some attributes that aren't used in here: `adj`, `target_shape`, `num_filter`, `workspace`, `no_bias`, and `layout`.

Also, the MXnet's Deconvolution op takes an optional "bias" input tensor, which I don't see being used in this code.

If this is all intentional, it would be helpful IMO to have some comments explaining why they're not needed.

Reply:
That's a good point, they're also not tested...